CONTRIBUTING.md

```diff
@@ -54,7 +52,151 @@ We have provided at profile for java located in [Codestyle File ./docs/CodeStyle
 - [Visual Studio Code](https://stackoverflow.com/questions/46030629/need-to-import-eclipse-java-formatter-profile-in-visual-studio-code)
 
-#### License
+## Commit Style
+
+SystemDS project has linear history throughout. Rebase cleanly and verify that the commit-SHA's
+of the Apache Repo are not altered.
+A general guideline is never to change a commit inside the systemds official repo, which is
+the standard practice. If you have accidentally committed a functionality or tag, let others know.
+And create a new commit to revert the functionality of the previous commit but **not force push**.
+
+
+### Tags
+
+The tags can be used in combination to one another. These are the only tags available.
+
+* `[MINOR]`: Small changesets with additional functionality
+
+  > Examples:
+  >
+  > This commit makes small software updates with refactoring
+  > 
+  > [`030fdab3`](https://github.com/apache/systemds/commit/030fdab3ebe6dedc3b4bb860e0ec5acfd9c38e5d) - `[MINOR] Added package to R dependency; updated Docker test image`
+  >
+  > This commit cleans up the redundant code for simplification
+  > 
+  > [`f4fa5650`](https://github.com/apache/systemds/commit/f4fa565013de13270df05dd37610382ca80f7354) - `[MINOR][SYSTEMDS-43] Cleanup scale builtin function (readability)`
+  >
+
+* `[SYSTEMDS-#]`: A changeset which will a specific purpose such as a bug, Improvement, 
+   New feature, etc. The tag value is found from [SystemDS jira issue tracker](https://issues.apache.org/jira/projects/SYSTEMDS/issues).
+   
+* `[DOC]` also `[DOCS]`: Changes to the documentation
+  
+* `[HOTFIX]`: Introduce a fix in the _hot_ running system.
+    
+    > Example:
+    > 
+    > This commit fixes the corrupted language path
+    > 
+    > [`87bc3584`](https://github.com/apache/systemds/commit/87bc3584db2148cf78b2d46418639e88ca27ec64) - `[HOTFIX] Fix validation of scalar-scalar binary min/max operations`
+    >
+
+### Commit description
+
+> A commit or PR description is a public record of **what** change is being made and **why**.
+
+#### Structure of the description
+
+##### First Line
+
+1. A summary of what the changeset.
+2. A complete sentence, crafted as though it was an order.
+    - an imperative sentence
+    - Writing the rest of the description as an imperative is optional.
+3. Follow by an empty line.
+
+##### Body
+
+It consists of the following.
+
+1. A brief description of the problem solved.
+2. Why this is the best approach?.
+3. Shortcomings to the approach, if any (important!).
+
+Additional info
+
+4. background information
+   - bug/issue/jira numbers
+   - benchmark/test results
+   - links to design documents
+   - code attributions
+5. Include enough context for
+   - reviewers
+   - future readers to understand the Changes.
+6. Add PR number, like `Closes #1000`.
+
+The following is a commit description with all the points mentioned.
+
+[`1abe9cb`](https://github.com/apache/systemds/commit/1abe9cb79d8001992f1c79ba5e638e6b423a1382)
+
+```
```

Comment:
It is good style to specify what type of code block you have in markdown, here in this instance it is txt